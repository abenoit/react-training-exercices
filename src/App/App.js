import React, { Component } from 'react';
import { PageHeader } from 'react-bootstrap';
import Cat from '../Cat/Cat';
import CatForm from '../Cat/CatForm';
import './App.css';

class App extends Component {
  constructor() {
    super();

    this.state = {
      cats: []
    };

    this.addCat = this.addCat.bind(this);
  }

  componentDidMount() {
    fetch('./cats.json')
      .then(res => res.json())
      .then(data => {
        this.setState({ cats: data });
      });
  }

  addCat(cat) {
    this.setState({ cats: [...this.state.cats, cat] });
  }

  render() {
    return (
      <div className="App">
        <PageHeader>
          React Training <small>stage 7</small>
        </PageHeader>

        <CatForm addCat={this.addCat} />

        {this.state.cats.length > 0 ? (
          this.state.cats.map(cat => (
            <Cat
              key={cat.name}
              name={cat.name}
              age={cat.age}
              meowsPerHour={cat.meowsPerHour}
              color={cat.color}
              meow={() => {
                console.log('meow from app');
              }}
            />
          ))
        ) : (
          <div
            style={{ padding: '20px 0', fontSize: '2em', color: 'deeppink' }}
          >
            No cat to display
          </div>
        )}
      </div>
    );
  }
}

export default App;
