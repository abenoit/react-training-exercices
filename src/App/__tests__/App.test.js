import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';

describe('App', () => {
  it('renders without crashing', () => {
    window.fetch = jest
      .fn()
      .mockImplementation(() =>
        Promise.resolve({ json: () => Promise.resolve([]) })
      );

    const app = renderer.create(<App />).toJSON();
    expect(app).toMatchSnapshot();
  });
});
