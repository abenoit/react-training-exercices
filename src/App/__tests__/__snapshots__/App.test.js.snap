// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<div
  className="App"
>
  <div
    className="page-header"
  >
    <h1>
      React Training 
      <small>
        stage 7
      </small>
    </h1>
  </div>
  <div
    className="CatForm"
  >
    <h2>
       Add a new cat 
    </h2>
    <form
      className="form-inline"
    >
      <label
        className="control-label"
        htmlFor={undefined}
      >
        Name * 
      </label>
      <input
        className="form-control"
        id={undefined}
        onChange={[Function]}
        placeholder="Name"
        type={undefined}
        value=""
      />
      <label
        className="control-label"
        htmlFor={undefined}
      >
        Age * 
      </label>
      <input
        className="form-control"
        id={undefined}
        onChange={[Function]}
        placeholder="Age"
        type="number"
        value={1}
      />
      <label
        className="control-label"
        htmlFor={undefined}
      >
        MeowsPerHour
      </label>
      <input
        className="form-control"
        id={undefined}
        onChange={[Function]}
        placeholder="MeowsPerHour"
        type="number"
        value={10}
        width="200px"
      />
      <label
        className="control-label"
        htmlFor={undefined}
      >
        Color
      </label>
      <input
        className="form-control"
        id={undefined}
        onChange={[Function]}
        placeholder="Color"
        type={undefined}
        value="green"
        width="200px"
      />
      <button
        className="btn btn-primary"
        disabled={true}
        onClick={[Function]}
        type="button"
      >
        Add this cat
      </button>
    </form>
  </div>
  <div
    style={
      Object {
        "color": "deeppink",
        "fontSize": "2em",
        "padding": "20px 0",
      }
    }
  >
    No cat to display
  </div>
</div>
`;
